File: LeetCode/Easy/704. Binary Search.py
class Solution:
    def search( self, nums: List[int], target: int ) -> int:
        p, q = 0, len( nums ) - 1
        
        while p < q:
            i = ( p + q ) // 2
            
            if nums[i] < target:
                p = i + 1
            else:
                q = i
                
        return p if nums[p] == target else -1

File: LeetCode/Easy/704. Binary Search.php
class Solution
{



    function search( $nums, $target )
    {
        $i = 0;
        $j = count( $nums ) - 1;

        while ( $i <= $j )
        {
            $k = floor( ( $i + $j ) / 2 );
            
            if ( $nums[$k] < $target )
            {
                $i = $k + 1;
            }
            else
            {
                $j = $k - 1;
            }
        }

        if ( $nums[$i] == $target )
        {
            return $i;
        }
        else
        {
            return -1;
        }
    }
}
